sql/plan/hash_lookup.go

```diff
@@ -137,6 +139,10 @@ func (n *HashLookup) GetHashKey(ctx *sql.Context, e sql.Expression, row sql.Row)
 	if k, ok := key.([]byte); ok {
 		key = string(k)
 	}
+	// decimals are not hashable
+	if d, ok := key.(decimal.Decimal); ok {
+		key = d.String()
```

Comment:
I believe we handle something similar when hashing values for Tuples.
See `hashOfSimple` in `gms/sql/expression.in.go`.